File: tests/criterion/php/support/numbering_system_test.c
#include "src/php/support/numbering_system.h"
#include "tests/criterion/test.h"

#define TEST_SUITE phpSupportNumberingSystem

ParameterizedTestParameters(TEST_SUITE, getsNumberingSystemCaseNameForValue) {
  START_BASIC_TEST_PARAMS(92)

  BASIC_TEST("adlm", "Adlm")
  BASIC_TEST("ahom", "Ahom")
  BASIC_TEST("arab", "Arab")
  BASIC_TEST("arabext", "Arabext")
  BASIC_TEST("armn", "Armn")
  BASIC_TEST("armnlow", "Armnlow")
  BASIC_TEST("bali", "Bali")
  BASIC_TEST("beng", "Beng")
  BASIC_TEST("bhks", "Bhks")
  BASIC_TEST("brah", "Brah")
  BASIC_TEST("cakm", "Cakm")
  BASIC_TEST("cham", "Cham")
  BASIC_TEST("cyrl", "Cyrl")
  BASIC_TEST("deva", "Deva")
  BASIC_TEST("diak", "Diak")
  BASIC_TEST("ethi", "Ethi")
  BASIC_TEST("finance", "Finance")
  BASIC_TEST("fullwide", "Fullwide")
  BASIC_TEST("geor", "Geor")
  BASIC_TEST("gong", "Gong")
  BASIC_TEST("gonm", "Gonm")
  BASIC_TEST("grek", "Grek")
  BASIC_TEST("greklow", "Greklow")
  BASIC_TEST("gujr", "Gujr")
  BASIC_TEST("guru", "Guru")
  BASIC_TEST("hanidays", "Hanidays")
  BASIC_TEST("hanidec", "Hanidec")
  BASIC_TEST("hans", "Hans")
  BASIC_TEST("hansfin", "Hansfin")
  BASIC_TEST("hant", "Hant")
  BASIC_TEST("hantfin", "Hantfin")
  BASIC_TEST("hebr", "Hebr")
  BASIC_TEST("hmng", "Hmng")
  BASIC_TEST("hmnp", "Hmnp")
  BASIC_TEST("java", "Java")
  BASIC_TEST("jpan", "Jpan")
  BASIC_TEST("jpanfin", "Jpanfin")
  BASIC_TEST("jpanyear", "Jpanyear")
  BASIC_TEST("kali", "Kali")
  BASIC_TEST("kawi", "Kawi")
  BASIC_TEST("khmr", "Khmr")
  BASIC_TEST("knda", "Knda")
  BASIC_TEST("lana", "Lana")
  BASIC_TEST("lanatham", "Lanatham")
  BASIC_TEST("laoo", "Laoo")
  BASIC_TEST("latn", "Latn")
  BASIC_TEST("lepc", "Lepc")
  BASIC_TEST("limb", "Limb")
  BASIC_TEST("mathbold", "Mathbold")
  BASIC_TEST("mathdbl", "Mathdbl")
  BASIC_TEST("mathmono", "Mathmono")
  BASIC_TEST("mathsanb", "Mathsanb")
  BASIC_TEST("mathsans", "Mathsans")
  BASIC_TEST("mlym", "Mlym")
  BASIC_TEST("modi", "Modi")
  BASIC_TEST("mong", "Mong")
  BASIC_TEST("mroo", "Mroo")
  BASIC_TEST("mtei", "Mtei")
  BASIC_TEST("mymr", "Mymr")
  BASIC_TEST("mymrshan", "Mymrshan")
  BASIC_TEST("mymrtlng", "Mymrtlng")
  BASIC_TEST("nagm", "Nagm")
  BASIC_TEST("native", "Native")
  BASIC_TEST("newa", "Newa")
  BASIC_TEST("nkoo", "Nkoo")
  BASIC_TEST("olck", "Olck")
  BASIC_TEST("orya", "Orya")
  BASIC_TEST("osma", "Osma")
  BASIC_TEST("rohg", "Rohg")
  BASIC_TEST("roman", "Roman")
  BASIC_TEST("romanlow", "Romanlow")
  BASIC_TEST("saur", "Saur")
  BASIC_TEST("segment", "Segment")
  BASIC_TEST("shrd", "Shrd")
  BASIC_TEST("sind", "Sind")
  BASIC_TEST("sinh", "Sinh")
  BASIC_TEST("sora", "Sora")
  BASIC_TEST("sund", "Sund")
  BASIC_TEST("takr", "Takr")
  BASIC_TEST("talu", "Talu")
  BASIC_TEST("taml", "Taml")
  BASIC_TEST("tamldec", "Tamldec")
  BASIC_TEST("telu", "Telu")
  BASIC_TEST("thai", "Thai")
  BASIC_TEST("tibt", "Tibt")
  BASIC_TEST("tirh", "Tirh")
  BASIC_TEST("tnsa", "Tnsa")
  BASIC_TEST("traditio", "Traditio")
  BASIC_TEST("vaii", "Vaii")
  BASIC_TEST("wara", "Wara")
  BASIC_TEST("wcho", "Wcho")
  BASIC_TEST("foo", NULL)
  BASIC_TEST(NULL, NULL)

  END_BASIC_TEST_PARAMS;
}

BASIC_PARAMETERIZED_TEST(getsNumberingSystemCaseNameForValue,
                         getNumberingSystemCaseName)

File: tests/phpunit/Intl/NumberingSystemTest.php
<?php

namespace Ecma\Test\Intl;

use Ecma\Intl\NumberingSystem;
use Ecma\Test\TestCase;

class NumberingSystemTest extends TestCase
{
    /**
     * @dataProvider numberingSystemValuesProvider
     */
    public function testNumberingSystemValues(NumberingSystem $numberingSystem, string $expectedValue): void
    {
        $this->assertSame($expectedValue, $numberingSystem->value);
    }

    /**
     * @return array<array{numberingSystem: NumberingSystem, expectedValue: string}>
     */
    public function numberingSystemValuesProvider(): array
    {
        return [
            [
                'numberingSystem' => NumberingSystem::Adlm,
                'expectedValue' => 'adlm',
            ],
            [
                'numberingSystem' => NumberingSystem::Ahom,
                'expectedValue' => 'ahom',
            ],
            [
                'numberingSystem' => NumberingSystem::Arab,
                'expectedValue' => 'arab',
            ],
            [
                'numberingSystem' => NumberingSystem::Arabext,
                'expectedValue' => 'arabext',
            ],
            [
                'numberingSystem' => NumberingSystem::Armn,
                'expectedValue' => 'armn',
            ],
            [
                'numberingSystem' => NumberingSystem::Armnlow,
                'expectedValue' => 'armnlow',
            ],
            [
                'numberingSystem' => NumberingSystem::Bali,
                'expectedValue' => 'bali',
            ],
            [
                'numberingSystem' => NumberingSystem::Beng,
                'expectedValue' => 'beng',
            ],
            [
                'numberingSystem' => NumberingSystem::Bhks,
                'expectedValue' => 'bhks',
            ],
            [
                'numberingSystem' => NumberingSystem::Brah,
                'expectedValue' => 'brah',
            ],
            [
                'numberingSystem' => NumberingSystem::Cakm,
                'expectedValue' => 'cakm',
            ],
            [
                'numberingSystem' => NumberingSystem::Cham,
                'expectedValue' => 'cham',
            ],
            [
                'numberingSystem' => NumberingSystem::Cyrl,
                'expectedValue' => 'cyrl',
            ],
            [
                'numberingSystem' => NumberingSystem::Deva,
                'expectedValue' => 'deva',
            ],
            [
                'numberingSystem' => NumberingSystem::Diak,
                'expectedValue' => 'diak',
            ],
            [
                'numberingSystem' => NumberingSystem::Ethi,
                'expectedValue' => 'ethi',
            ],
            [
                'numberingSystem' => NumberingSystem::Finance,
                'expectedValue' => 'finance',
            ],
            [
                'numberingSystem' => NumberingSystem::Fullwide,
                'expectedValue' => 'fullwide',
            ],
            [
                'numberingSystem' => NumberingSystem::Geor,
                'expectedValue' => 'geor',
            ],
            [
                'numberingSystem' => NumberingSystem::Gong,
                'expectedValue' => 'gong',
            ],
            [
                'numberingSystem' => NumberingSystem::Gonm,
                'expectedValue' => 'gonm',
            ],
            [
                'numberingSystem' => NumberingSystem::Grek,
                'expectedValue' => 'grek',
            ],
            [
                'numberingSystem' => NumberingSystem::Greklow,
                'expectedValue' => 'greklow',
            ],
            [
                'numberingSystem' => NumberingSystem::Gujr,
                'expectedValue' => 'gujr',
            ],
            [
                'numberingSystem' => NumberingSystem::Guru,
                'expectedValue' => 'guru',
            ],
            [
                'numberingSystem' => NumberingSystem::Hanidays,
                'expectedValue' => 'hanidays',
            ],
            [
                'numberingSystem' => NumberingSystem::Hanidec,
                'expectedValue' => 'hanidec',
            ],
            [
                'numberingSystem' => NumberingSystem::Hans,
                'expectedValue' => 'hans',
            ],
            [
                'numberingSystem' => NumberingSystem::Hansfin,
                'expectedValue' => 'hansfin',
            ],
            [
                'numberingSystem' => NumberingSystem::Hant,
                'expectedValue' => 'hant',
            ],
            [
                'numberingSystem' => NumberingSystem::Hantfin,
                'expectedValue' => 'hantfin',
            ],
            [
                'numberingSystem' => NumberingSystem::Hebr,
                'expectedValue' => 'hebr',
            ],
            [
                'numberingSystem' => NumberingSystem::Hmng,
                'expectedValue' => 'hmng',
            ],
            [
                'numberingSystem' => NumberingSystem::Hmnp,
                'expectedValue' => 'hmnp',
            ],
            [
                'numberingSystem' => NumberingSystem::Java,
                'expectedValue' => 'java',
            ],
            [
                'numberingSystem' => NumberingSystem::Jpan,
                'expectedValue' => 'jpan',
            ],
            [
                'numberingSystem' => NumberingSystem::Jpanfin,
                'expectedValue' => 'jpanfin',
            ],
            [
                'numberingSystem' => NumberingSystem::Jpanyear,
                'expectedValue' => 'jpanyear',
            ],
            [
                'numberingSystem' => NumberingSystem::Kali,
                'expectedValue' => 'kali',
            ],
            [
                'numberingSystem' => NumberingSystem::Kawi,
                'expectedValue' => 'kawi',
            ],
            [
                'numberingSystem' => NumberingSystem::Khmr,
                'expectedValue' => 'khmr',
            ],
            [
                'numberingSystem' => NumberingSystem::Knda,
                'expectedValue' => 'knda',
            ],
            [
                'numberingSystem' => NumberingSystem::Lana,
                'expectedValue' => 'lana',
            ],
            [
                'numberingSystem' => NumberingSystem::Lanatham,
                'expectedValue' => 'lanatham',
            ],
            [
                'numberingSystem' => NumberingSystem::Laoo,
                'expectedValue' => 'laoo',
            ],
            [
                'numberingSystem' => NumberingSystem::Latn,
                'expectedValue' => 'latn',
            ],
            [
                'numberingSystem' => NumberingSystem::Lepc,
                'expectedValue' => 'lepc',
            ],
            [
                'numberingSystem' => NumberingSystem::Limb,
                'expectedValue' => 'limb',
            ],
            [
                'numberingSystem' => NumberingSystem::Mathbold,
                'expectedValue' => 'mathbold',
            ],
            [
                'numberingSystem' => NumberingSystem::Mathdbl,
                'expectedValue' => 'mathdbl',
            ],
            [
                'numberingSystem' => NumberingSystem::Mathmono,
                'expectedValue' => 'mathmono',
            ],
            [
                'numberingSystem' => NumberingSystem::Mathsanb,
                'expectedValue' => 'mathsanb',
            ],
            [
                'numberingSystem' => NumberingSystem::Mathsans,
                'expectedValue' => 'mathsans',
            ],
            [
                'numberingSystem' => NumberingSystem::Mlym,
                'expectedValue' => 'mlym',
            ],
            [
                'numberingSystem' => NumberingSystem::Modi,
                'expectedValue' => 'modi',
            ],
            [
                'numberingSystem' => NumberingSystem::Mong,
                'expectedValue' => 'mong',
            ],
            [
                'numberingSystem' => NumberingSystem::Mroo,
                'expectedValue' => 'mroo',
            ],
            [
                'numberingSystem' => NumberingSystem::Mtei,
                'expectedValue' => 'mtei',
            ],
            [
                'numberingSystem' => NumberingSystem::Mymr,
                'expectedValue' => 'mymr',
            ],
            [
                'numberingSystem' => NumberingSystem::Mymrshan,
                'expectedValue' => 'mymrshan',
            ],
            [
                'numberingSystem' => NumberingSystem::Mymrtlng,
                'expectedValue' => 'mymrtlng',
            ],
            [
                'numberingSystem' => NumberingSystem::Nagm,
                'expectedValue' => 'nagm',
            ],
            [
                'numberingSystem' => NumberingSystem::Native,
                'expectedValue' => 'native',
            ],
            [
                'numberingSystem' => NumberingSystem::Newa,
                'expectedValue' => 'newa',
            ],
            [
                'numberingSystem' => NumberingSystem::Nkoo,
                'expectedValue' => 'nkoo',
            ],
            [
                'numberingSystem' => NumberingSystem::Olck,
                'expectedValue' => 'olck',
            ],
            [
                'numberingSystem' => NumberingSystem::Orya,
                'expectedValue' => 'orya',
            ],
            [
                'numberingSystem' => NumberingSystem::Osma,
                'expectedValue' => 'osma',
            ],
            [
                'numberingSystem' => NumberingSystem::Rohg,
                'expectedValue' => 'rohg',
            ],
            [
                'numberingSystem' => NumberingSystem::Roman,
                'expectedValue' => 'roman',
            ],
            [
                'numberingSystem' => NumberingSystem::Romanlow,
                'expectedValue' => 'romanlow',
            ],
            [
                'numberingSystem' => NumberingSystem::Saur,
                'expectedValue' => 'saur',
            ],
            [
                'numberingSystem' => NumberingSystem::Segment,
                'expectedValue' => 'segment',
            ],
            [
                'numberingSystem' => NumberingSystem::Shrd,
                'expectedValue' => 'shrd',
            ],
            [
                'numberingSystem' => NumberingSystem::Sind,
                'expectedValue' => 'sind',
            ],
            [
                'numberingSystem' => NumberingSystem::Sinh,
                'expectedValue' => 'sinh',
            ],
            [
                'numberingSystem' => NumberingSystem::Sora,
                'expectedValue' => 'sora',
            ],
            [
                'numberingSystem' => NumberingSystem::Sund,
                'expectedValue' => 'sund',
            ],
            [
                'numberingSystem' => NumberingSystem::Takr,
                'expectedValue' => 'takr',
            ],
            [
                'numberingSystem' => NumberingSystem::Talu,
                'expectedValue' => 'talu',
            ],
            [
                'numberingSystem' => NumberingSystem::Taml,
                'expectedValue' => 'taml',
            ],
            [
                'numberingSystem' => NumberingSystem::Tamldec,
                'expectedValue' => 'tamldec',
            ],
            [
                'numberingSystem' => NumberingSystem::Telu,
                'expectedValue' => 'telu',
            ],
            [
                'numberingSystem' => NumberingSystem::Thai,
                'expectedValue' => 'thai',
            ],
            [
                'numberingSystem' => NumberingSystem::Tibt,
                'expectedValue' => 'tibt',
            ],
            [
                'numberingSystem' => NumberingSystem::Tirh,
                'expectedValue' => 'tirh',
            ],
            [
                'numberingSystem' => NumberingSystem::Tnsa,
                'expectedValue' => 'tnsa',
            ],
            [
                'numberingSystem' => NumberingSystem::Traditio,
                'expectedValue' => 'traditio',
            ],
            [
                'numberingSystem' => NumberingSystem::Vaii,
                'expectedValue' => 'vaii',
            ],
            [
                'numberingSystem' => NumberingSystem::Wara,
                'expectedValue' => 'wara',
            ],
            [
                'numberingSystem' => NumberingSystem::Wcho,
                'expectedValue' => 'wcho',
            ],
        ];
    }
}
